Automattic/wp-calypso
# Site Transfers: Prototype UX to initiate a site transfer

See https://github.com/Automattic/dotcom-forge/issues/2496

## Proposed Changes

Prototypes a workflow for a site owner to transfer their site to another WordPress.com user.

However, this will change substantially in a follow-up issue https://github.com/Automattic/dotcom-forge/issues/2496#issuecomment-1558897012

### New 'Transfer your site' option in general settings

<img width="767" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/629b8e44-5e4f-4535-b2c4-1a50ba4fa87f">

### Initial screen

<img width="855" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/144c2657-afb3-4129-818e-ce1f15c2fc18">

### Input appears once you've toggled the boxes

<img width="828" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/f4a25f94-719c-440f-9733-ea5acd7fdd63">

### Success

## Testing Instructions

1. Apply D111563-code to your sandbox
2. Create a new Simple site with a Personal plan using a test user.
3. Navigate to General Settings and click 'Transfer your site'.
4. Click on the toggles and verify the input appears.
5. Add an invalid username, hit submit, and verify an error message appears.
6. Add a username for a second test user, hit submit, and verify the success message appears.
7. Verify you've received a confirmation email in your inbox.
